packages/@aws-cdk/aws-servicecatalogappregistry/test/application.test.ts

```diff
@@ -0,0 +1,122 @@
+import '@aws-cdk/assert-internal/jest';
+import * as cdk from '@aws-cdk/core';
+import * as appreg from '../lib';
+
+describe('Application', () => {
+  let stack: cdk.Stack;
+
+  beforeEach(() => {
+    stack = new cdk.Stack();
+  });
+
+  test('default application creation', () => {
+    new appreg.Application(stack, 'MyApplication', {
+      applicationName: 'testApplication',
+    });
+
+    expect(stack).toMatchTemplate({
+      Resources: {
+        MyApplication5C63EC1D: {
+          Type: 'AWS::ServiceCatalogAppRegistry::Application',
+          Properties: {
+            Name: 'testApplication',
+          },
+        },
+      },
+    });
+  }),
+
+  test('application with explicit description', () => {
+    const description = 'my test application description';
+    new appreg.Application(stack, 'MyApplication', {
+      applicationName: 'testApplication',
+      description: description,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalogAppRegistry::Application', {
+      Description: description,
+    });
+  }),
+
+  test('application with application tags', () => {
+    const application = new appreg.Application(stack, 'MyApplication', {
+      applicationName: 'testApplication',
+    });
+
+    cdk.Tags.of(application).add('key1', 'value1');
+    cdk.Tags.of(application).add('key2', 'value2');
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalogAppRegistry::Application', {
+      Tags: {
+        key1: 'value1',
+        key2: 'value2',
+      },
+    });
+  }),
+
+  test('for an application imported by ARN', () => {
+    const application = appreg.Application.fromApplicationArn(stack, 'MyApplication',
+      'arn:aws:servicecatalog:us-east-1:123456789012:/applications/0aqmvxvgmry0ecc4mjhwypun6i');
+    expect(application.applicationId).toEqual('0aqmvxvgmry0ecc4mjhwypun6i');
+  }),
+
+  test('fails for application imported by ARN missing applicationId', () => {
+    expect(() => {
+      appreg.Application.fromApplicationArn(stack, 'MyApplication',
+        'arn:aws:servicecatalog:us-east-1:123456789012:/applications/');
+    }).toThrow(/Malformed ARN, cannot determine application ID from:/);
+  }),
+
+  test('application created with a token description does not throw validation error', () => {
+    const tokenDescription = cdk.Lazy.string({ produce: () => 'token description'.repeat(1000) });
+    expect(() => {
+      new appreg.Application(stack, 'MyApplication', {
+        applicationName: 'myApplication',
+        description: tokenDescription,
+      });
+    }).not.toThrow(/Invalid application description for resource/);
```

Comment:
Can you use a CloudFormation Parameter here instead of a Lazy? Using `Lazy` like that makes the test look weird (unrealistic).

I would also just assert on the template contents:

```suggestion
    const param = new cdk.CfnParameter(stack, 'Param');
    new appreg.Application(stack, 'MyApplication', {
      applicationName: 'myApplication',
      description: param.valueAsString,
    });
    
    expect(stack).toHaveResourceLike('AWS::ServiceCatalogAppRegistry::Application', {
      Description: {
        Ref: 'Param',
      }, 
    });
```

Looks like we're missing a similar test for `applicationName`, BTW.

Reply:
updated to test template for the fields, both name and description